libraries/Microsoft.Bot.Builder.LanguageGeneration/CustomizedMemory.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.Bot.Builder.LanguageGeneration
     /// we want to make sure the global memory (the first memory passed in) can be
     /// accessible at any sub evaluation process. 
     /// </summary>
-    internal class CustomizedMemory : IMemory
+    public class CustomizedMemory : IMemory
```

Comment:
I think we should still keep this private, instead, we should create a public method 'Evaluate' in evaluator, which accept a normal object, so that we will leak less to upper layer